fiftyone/core/dataset.py

```diff
@@ -2841,7 +2841,7 @@ def save_view(
         """
         if view._root_dataset is not self:
             raise ValueError("Cannot save view into a different dataset")
-
+        view._set_name(name)
```

Comment:
Note: We are only allowing the view.name to be set if the view is loaded from a saved view. Thus, the only way to get the name of a DatasetView to be a specific string is to call `dataset.save_view("<view_name>")` or `dataset.update_view_info("<current_view_name>", {"name": "<new_view_name>"}).

Reply:
Or `load_view()`! Which would be the typical way to get your hands on a saved view.

I don't think this is relevant to this PR, but an important consideration will be that accessing a saved view loaded from within the App via `session.view` should also have the saved view name in-place in Python

Reply:
I set the name on `load_view` [here](https://github.com/voxel51/fiftyone/pull/2218/files#diff-d4170c49dd9265f224bc766f1675ff3a6690cb4103051867110901c46f787abeR2950) as well, but the method name got collapsed in the diff. Is this what you were referring to?

Agree. I'm handling the App side of it in another PR

Reply:
Yeah no action, I was just commenting that the primary way a user will get their hands on a saved view is by calling `load_view()` or accessing `session.view` after loading a saved view in the App

Reply:
I see that you handled `load_view()` in your implementation. Nice 💪